cmd/api/src/api/v2/database_wipe.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v2
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gofrs/uuid"
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/dawgs/ops"
+	"github.com/specterops/bloodhound/log"
+	"github.com/specterops/bloodhound/src/api"
+	"github.com/specterops/bloodhound/src/model"
+)
+
+type DatabaseWipe struct {
+	DeleteCollectedGraphData bool `json:"deleteCollectedGraphData"`
+	DeleteHighValueSelectors bool `json:"deleteHighValueSelectors"`
+	DeleteFileIngestHistory  bool `json:"deleteFileIngestHistory"`
+	DeleteDataQualityHistory bool `json:"deleteDataQualityHistory"`
+	AssetGroupId             int  `json:"assetGroupId"`
+}
+
+func (s Resources) HandleDatabaseWipe(response http.ResponseWriter, request *http.Request) {
+
+	var (
+		payload DatabaseWipe
+		// use this struct to flag any fields that failed to delete
+		errors []string
+		// deleting collected graph data OR high value selectors starts analsyis
+		kickoffAnalysis bool
+	)
+
+	if err := api.ReadJSONRequestPayloadLimited(&payload, request); err != nil {
+		api.WriteErrorResponse(
+			request.Context(),
+			api.BuildErrorResponse(http.StatusBadRequest, "JSON malformed.", request),
+			response,
+		)
+		return
+	}
+
+	// return `BadRequest` if request specifies `high value selectors` without an `asset group ID`
+	if payload.DeleteHighValueSelectors && payload.AssetGroupId == 0 {
+		api.WriteErrorResponse(
+			request.Context(),
+			api.BuildErrorResponse(http.StatusBadRequest, "please provide an assetGroupId to delete", request),
+			response,
+		)
+		return
+	}
```

Comment:
Assuming the answer to my previous question is "the service should not need the requestor to provide an asset group ID" then we can remove this validation

Reply:
removed!